app/services/apply_mods_metadata.rb

```diff
@@ -66,28 +67,26 @@ def log_error!(exception)
     log.puts(exception.backtrace.to_s)
   end
 
-  # Open a new version for the given object if it is in the accessioned state.
-  def version_object
-    return unless accessioned?
-
-    unless DorObjectWorkflowStatus.new(cocina.externalIdentifier, version: cocina.version).can_open_version?
-      log.puts("argo.bulk_metadata.bulk_log_unable_to_version #{cocina.externalIdentifier}") # totally unexpected
-      return
-    end
-    commit_new_version
-  end
-
   def update_metadata
-    object_client = Dor::Services::Client.object(cocina.externalIdentifier)
+    object_client = Dor::Services::Client.object(updated_cocina.externalIdentifier)
     object_client.metadata.update_mods(mods)
```

Comment:
use the new object when updating metadata

Reply:
We're not using the updated object, just it's identifier.